handlers/showHandlers.js

```diff
@@ -0,0 +1,146 @@
+import Show from '../models/show';
+import { stringSearch } from '../helpers/stringSearch';
+
+const getShowById = async (req, res) => {
+  const { id } = req.params;
+  const show = await Show.find({ show_id: +id });
+
+  if (show.length > 0) {
+    res.json({
+      statusCode: 200,
+      message: 'Movie fetched successfully',
+      data: show
+    });
+  } else {
+    res.status(404).json({
+      statusCode: 404,
+      error: `No movie found with id ${id}`,
+      params: req.params
+    });
+  }
+};
+
+const getAllShows = (req, res) => {
+  const {
+    page,
+    director,
+    title,
+    actor,
+    country,
+    year,
+    rating,
+    duration,
+    category,
+    description,
+    type
+  } = req.query;
```

Comment:
Good destructuring!

Reply:
If you need access to many object properties desctructuring is definitely the way to go.